### PR TITLE
make all themes customizable

### PR DESCRIPTION
**Pull request by mshd**
TODO: persist changes, when changing themes

### CODE REVIEW

**Comment by mshd** [comment]
Ok, maybe we can insert a timer, whenever we change one var, it loads all nodes, etc...., not a good performance

File: src/App.js [hunk 1]
```diff
@@ -79,6 +85,9 @@ export default class App extends Component {
         if (next.parent !== prev.parent) return currentTheme.separationCousins;
       });
       this.setState({ currentTheme });
+      this.setState({
+        currentCustomTheme: currentTheme,
```
Comment (ogroppo) [inline src/App.js:89]: mmmh, I think what you want to do here is to set both `currentTheme` and `currentCustomTheme`

File: src/hooks/useLoadTheme.js [hunk 2]
```diff
@@ -7,8 +7,14 @@ const useLoadTheme = () => {
...
   useEffect(() => {
+    //Clear all local storage for updates...
+    const currentVersion = 1;
+    if (!ls("lsVersion") || ls("lsVersion") !== currentVersion) {
+      ls.clear();
+      ls("lsVersion", currentVersion);
+    }
     const storedThemeKey = ls("storedThemeKey");
-    const storedCustomTheme = ls("storedCustomTheme");
+    const storedCustomTheme = ls("storedCustomTheme_" + storedThemeKey);
```
Comment (ogroppo) [inline src/hooks/useLoadTheme.js:17]:
```suggestion
    const storedCustomTheme = storedThemeKey ? ls("storedCustomTheme_" + storedThemeKey) : undefined;
```

File: src/hooks/useLoadTheme.js [hunk 3]
```diff
@@ -19,7 +25,7 @@ const useLoadTheme = () => {
     if (!storedThemeKey) {
       setCurrentTheme(defaultTheme, false);
     } else {
-      if (storedThemeKey === "Custom") {
+      if (!storedCustomTheme) {
```
Comment (ogroppo) [inline src/hooks/useLoadTheme.js:28]: Should be the other way round, no?
```suggestion
      if (storedCustomTheme) {
```

File: src/modals/SettingsModal/CustomThemeForm.js [hunk 4]
```diff
@@ -341,13 +352,40 @@ export default function CustomThemeForm() {
...
+        {/*      label="hide"*/}
+        {/*      name="isInIframe"*/}
+        {/*      id="isInIframe-hide"*/}
+        {/*      checked={currentCustomTheme.isInIframe === true}*/}
+        {/*      onChange={(e) =>*/}
+        {/*        setCustomThemeProp("isInIframe", true)*/}
+        {/*      }*/}
+        {/*    />*/}
+        {/*  </Col>*/}
+        {/*</Form.Group>*/}
```
Comment (ogroppo) [inline src/modals/SettingsModal/CustomThemeForm.js:388]: ?

File: src/modals/SettingsModal/SettingsModal.js [hunk 5]
```diff
@@ -52,19 +61,33 @@ export default function SettingsModal({ show, hideModal }) {
                 eventKey={index + 1}
                 active={theme.name === currentTheme.name}
                 disabled={theme.disabled}
-                onClick={() =>
-                  setCurrentTheme(theme.isCustom ? customTheme : theme)
-                }
+                onClick={() => setCurrentTheme(theme)}
```
Comment (ogroppo) [inline src/modals/SettingsModal/SettingsModal.js:64]: better to put here the logic 
```
//set currentTheme using previously locally stored theme
      const storedTheme = ls("storedCustomTheme_" + currentTheme.name);
     setCurrentTheme(storedTheme | theme)
```

File: src/modals/SettingsModal/SettingsModal.js [hunk 6]
```diff
@@ -52,19 +61,33 @@ export default function SettingsModal({ show, hideModal }) {
...
               >
                 {theme.name}
               </Dropdown.Item>
             ))}
-          </Dropdown.Menu>
+          </Dropdown.Menu>{" "}
+          customize it{" "}
+          <a
```
Comment (ogroppo) [inline src/modals/SettingsModal/SettingsModal.js:71]: Just a reminder to turn this into a button link (empty href can be an issue)

**Review by ogroppo — CHANGES_REQUESTED** [review]

File: src/modals/SettingsModal/CustomThemeForm.js (hunk 4 above)
Comment (mshd) [reply src/modals/SettingsModal/CustomThemeForm.js:388]: I was thinking to hide the headers, but once there hidden, how to show them again😂😂😂

File: src/hooks/useLoadTheme.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ const useLoadTheme = () => {
   const { setCurrentTheme, setCustomTheme } = useContext(AppContext);
 
   useEffect(() => {
-    const currentLsVersion = 1;
+    const currentLsVersion = 2;
```
Comment (ogroppo) [inline src/hooks/useLoadTheme.js:10]: this has never been deployed, is there a need to increase?

File: src/modals/SettingsModal/CustomThemeForm.js [hunk 8]
```diff
@@ -566,6 +566,29 @@ export default function CustomThemeForm() {
           </Col>
         </Form.Group>
       </fieldset>
+      <fieldset>
+        <Form.Group as={Row}>
+          <Form.Label as="legend" column sm={3} className="pt-0">
+            More
+          </Form.Label>
+          <Col sm={9}>
+            <Form.Group controlId={"relStroke"}>
```
Comment (ogroppo) [inline src/modals/SettingsModal/CustomThemeForm.js:575]:
```suggestion
            <Form.Group controlId={"nodeCss"}>
```

**Review by ogroppo — CHANGES_REQUESTED** [review]

**Comment by ogroppo** [comment]
Also, someone could do CSS injection, but it's only run on the client so they will just harm themselves

File: src/hooks/useLoadTheme.js (hunk 7 above)
Comment (mshd) [reply src/hooks/useLoadTheme.js:10]: you're right

**Pull request merged**